### ISSUE TITLE
Wishlist: provide a toggle to prevent the creation of .ipynb_checkpoints

### ISSUE THREAD

**Issue by callegar** [jupyterlab/jupyterlab#11826]
### Problem

Checkpoints can be quite useful, but not in every context. When you just want to quickly view or run a notebook, they can become a nuisance polluting your directories.  Furthermore, if you manage your notebooks with a vcs they become a duplication of a functionality that is achieved in a different way.

Would be nice to be able to disable them with a toggle on the jupyterlab interface.

**Comment by hanmala123** [comment]
Each time I open a file(like check pictures in datasets), the checkpoints folder is created automatically in the dataset folder.
This always disrupt my dataset.

**Comment by krassowski** [comment]
Related to https://github.com/jupyterlab/jupyterlab/issues/4557 which proposes to auto-remove the `.ipynb_checkpoints` after tab closure.

**Comment by bollwyvl** [comment]
The `notebook` documentation includes an example of implementing a `NoOpCheckpoints`:

https://jupyter-notebook.readthedocs.io/en/stable/extending/contents.html#no-op-example

I guess an approach might be to package that and make it something that can be installed.

Another would be to check it into a `jupyter_config.py`.

**Comment by tristanbrown** [comment]
The .ipynb_checkpoints feature actually breaks Plotly Dash apps by polluting the /pages directory with duplicate files (all of which dash attempts to register as app pages). 

We really need a good way to disable this. I turned off autosave, but that doesn't stop the creation of these .ipynb_checkpoints directories.

**Comment by tristanbrown** [comment]
The .ipynb_checkpoints feature actually breaks Plotly Dash apps by polluting the /pages directory with duplicate files (all of which dash attempts to register as app pages). 

We really need a good way to disable this. I turned off autosave, but that doesn't stop the creation of these .ipynb_checkpoints directories.

**Comment by callegar** [comment]
A quick and dirty solution would be to allow one to create an empty `.ipynb_checkpoints` *file* in those dirs where he does not want an `.ipynb_checkpoints` dir to be autocreated and autopopulated. This would allow to control checkpoint creation per directory rather than globally.

Ideally, the checkpoint creation toggle should be gui-controllable and per .ipynb file, though.

**Comment by Walstruzz** [comment]
Any Update?

**Comment by callegar** [comment]
Yes, now we have another hidden item that after the shutdown remains as a leftover. I believe it is to support language servers ;-(

**Comment by trianta2** [comment]
The creation of `.ipynb_checkpoints` directories caused issues with MLflow. We need an easy way to disable this functionality.

**Comment by hutch3232** [comment]
@trianta2 maybe this isn't the most elegant solution but I followed this: https://github.com/jupyterlab/jupyterlab/issues/5809

I personally just threw everything into `/tmp`
```python
c.FileCheckpoints.checkpoint_dir=/tmp
```

N.B. it only uses filenames so duplicate filenames would cause overwriting. I personally do not care about this functionality at all because I use git.

**Comment by LBJ6666** [comment]
Has this feature been added？How should I set it up?

**Comment by sloisel** [comment]
I think the "Ju" in "Jupyter" stands for Julia? The official way of documenting Julia packages and code is [Documenter](https://github.com/JuliaDocs/Documenter.jl). The hidden `.ipynb_checkpoints` directories break Documenter.

**Comment by shawnbrown** [comment]
> The `notebook` documentation includes an example of implementing a `NoOpCheckpoints`:
> 
> https://jupyter-notebook.readthedocs.io/en/stable/extending/contents.html#no-op-example
>

The no-op example was removed starting in the 7.0.0 documentation. Here's a more persistent URL:

https://jupyter-notebook.readthedocs.io/en/v6.5.4/extending/contents.html#no-op-example
